src/js/map/TerrainGrid.js

```diff
@@ -0,0 +1,140 @@
+import Grid from './Grid'
+
+export default class TerrainGrid extends Grid {
+  constructor(tilesWide, tilesHigh) {
+    super(tilesWide, tilesHigh, 1)
+    this.setTerrainProperties()
+    this.reset()
+  }
+
+  setTerrainProperties() {
+    this.terrainProperties = {
+      normal: {
+        difficulty: 1,
+        color: 0xFFFFFF,
+      },
+      crater: {
+        difficulty: 3,
+        color: 0xCECECE,
+      },
+      ridge: {
+        difficulty: 2,
+        color: 0xE5E5E5,
+      },
+      obstacle: {
+        difficulty: 0,
+        color: 0x000000,
+      },
+    }
+  }
+
+  getTerrainProperties(type) {
+    if (this.terrainProperties.hasOwnProperty(type)) {
+      return {
+        type: type,
+        difficulty: this.terrainProperties[type].difficulty,
+        color: this.terrainProperties[type].color,
+      }
+    }
+    return {
+      type: "normal",
+      difficulty: this.terrainProperties.normal.difficulty,
+      color: this.terrainProperties.normal.color,
+    }
+  }
+
+  reset() {
+    this.initialValue = this.getTerrainProperties("normal")
+    super.reset()
+    this.generateTerrain()
+  }
+
+  generateTerrain() {
+    // TODO: Simple randomizer for terrain features
+    this.addTestCrater({x:17,y:17}, 4)
+    this.addTestCrater({x:30,y:30}, 4)
+    this.addTestCrater({x:30,y:4}, 4)
+  }
+
+  addTestCrater(gridLocation, size) {
+    // square crater
+    for (let x = gridLocation.x - size; x <= gridLocation.x + size; x++) {
+      for (let y = gridLocation.y - size; y <= gridLocation.y + size; y++) {
+        if (!this.coordinateIsValid(x, y)) { continue }
+        if (x == gridLocation.x - size || x == gridLocation.x + size ||
+            y == gridLocation.y - size || y == gridLocation.y + size) {
+          this.set(x, y, this.getTerrainProperties("ridge"))
+        } else if (x == gridLocation.x && y == gridLocation.y) {
+          this.set(x, y, this.getTerrainProperties("obstacle"))
+        } else {
+          this.set(x, y, this.getTerrainProperties("crater"))
+        }
+      }
+    }
+  }
+
+  randomize(wallProbability = 0.1) {
```

Comment:
Is this still being used? If not, let's remove it!

Reply:
You know how I love deleting things!